src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -261,12 +265,7 @@ private Map<Integer, Float> doANNSearch(final LeafReaderContext context, final B
                 new NativeMemoryEntryContext.IndexEntryContext(
                     indexPath.toString(),
                     NativeMemoryLoadStrategy.IndexLoadStrategy.getInstance(),
-                    getParametersAtLoading(
-                        spaceType,
-                        knnEngine,
-                        knnQuery.getIndexName(),
-                        FieldInfoExtractor.getIndexDescription(fieldInfo)
-                    ),
+                    getParametersAtLoading(spaceType, knnEngine, knnQuery.getIndexName(), indexDescription),
```

Comment:
Whyd we change this?

Reply:
Update: we updated to use vectorDataType instead of indexDescription.
```
getParametersAtLoading(spaceType, knnEngine, knnQuery.getIndexName(), vectorDataType),
```
When model exists, we get its value by
```
 vectorDataType = modelMetadata.getVectorDataType();
```